internal/statestore/redis.go

```diff
@@ -139,6 +200,36 @@ func (rb *redisBackend) HealthCheck(ctx context.Context) error {
 	return nil
 }
 
+func testOnBorrow(c redis.Conn, lastUsed time.Time) error {
+	// Assume the connection is valid if it was used in 30 sec.
+	if time.Since(lastUsed) < 15*time.Second {
+		return nil
+	}
+
+	_, err := c.Do("PING")
+	return err
+}
+
+func redisURLFromAddr(addr string, cfg config.View) string {
```

Comment:
nice to have - similar one line helper to get sentinel url - wont be long - but good to abstract into helper.